use-cases/using-vsocks.md

```diff
@@ -0,0 +1,133 @@
+# Kata Containers and vsocks
+
+- [Introduction](#introduction)
+    - [proxy communication diagram](#proxy-communication-diagram)
+    - [vsock communication diagram](#vsock-communication-diagram)
+- [System requirements](#system-requirements)
+- [Advantages of using vsocks](#advantages-of-using-vsocks)
+    - [High density](#high-density)
+    - [Reliability](#reliability)
+
+## Introduction
+
+There are two different ways to communicate processes in the virtual machine with
+processes in the host. The first one is by using serial ports, the processes in
+the virtual machine can read/write data from/to a serial port device and the
+processes in the host can read/write data from/to a unix socket. This solution
+has been part of QEMU from long time ago and almost all GNU/Linux distributions
+have support for serial ports, making it the most portable solution. However,
+there is a limitation that impact in its usage and implementation, the
+communication between processes is serial, making impossible that two or more
+processes read/write data at the same time. To deal with this limitation the
+resources (serial port and unix socket) must be multiplexed. In Kata Containers
+those resources are multiplexed by using [kata-proxy][2] and [yamux][3], the
+following diagram shows how it's implemented.
+
+
+### proxy communication diagram
+
+```
+.----------------------.
+| .------------------. |
+| | .-----.  .-----. | |
+| | |cont1|  |cont2| | |
+| | `-----'  `-----' | |
+| |       \   /      | |
+| |    .---------.   | |
+| |    |  agent  |   | |
+| |    `---------'   | |
+| |         |        | |
+| |    .-----------. | |
+| |POD |serial port| | |
+| `----|-----------|-' |
+|      |  socket   |   |
+|      `-----------'   |
+|           |          |
+|       .-------.      |
+|       | proxy |      |
+|       `-------'      |
+|           |          |
+|  .------./ \.------. |
+|  | shim |   | shim | |
+|  `------'   `------' |
+| Host                 |
+`----------------------'
+```
+
+The seconds way to communicate processes and the newer one is by using
```

Comment:
s/seconds/second/
s/to communicate process/for processes to communicate, and the newer one,/